db/test/versions/0008_test.js

```diff
@@ -0,0 +1,197 @@
+const _ = require('lodash');
+const helper = require('../helper');
+const testing = require('taskcluster-lib-testing');
+const assert = require('assert').strict;
+
+const ASCII = _.range(1, 128).map(i => String.fromCharCode(i)).join(' ');
+
+// (copied from azure-entities)
+const encodeStringKey = function(str) {
+  if (str === '') {
+    return '!';
+  }
+  return encodeURIComponent(str)
+    .replace(/!/g, '%21')
+    .replace(/~/g, '%7e')
+    .replace(/%/g, '!');
+};
+
+const decodeStringKey = function(str) {
+  if (str === '!') {
+    return '';
+  }
+  return decodeURIComponent(str.replace(/!/g, '%'));
```

Comment:
it took me a minute to realize we only need to do the `! -> %` replacement because `decodeURIComponent` will get the replacements for `!` and `~`. Maybe a comment explaining the seeming asymmetry between `encodeStringKey` and `decodeStringKey` could be nice.